src/api/article/article.model.js

```diff
@@ -46,119 +44,88 @@ const joiArticleSchema = Joi.object({
   // Images are as described in 'covers' guide by Vitalik.
   images: Joi.object().required(),
   // Can only be present when Article type is video.
-  video: joiVideoSchema,
-  color: Joi.color().default('#000000'),
+  video: Joi.object({
+    platform: Joi.string().valid(['youtube']),
+    id: Joi.string().regex(/^[a-zA-Z0-9_-]{11}$/),
+    url: Joi.string().uri(),
+  }),
+  audio: Joi.object({
+    platform: Joi.string().valid(['soundcloud']),
+    id: Joi.string(),
+    url: Joi.string().uri(),
+  }),
+  color: Joi.color().default(DEFAULT_COLOR),
   // Text on article card may be rendered in one of the following ways.
   // This depends on the color and is set manually.
-  textColorTheme: Joi.string()
+  theme: Joi.string()
     .valid(['light', 'dark'])
-    .default('light'),
+    .default(DEFAULT_THEME),
   // Authors and Brands are also just Tags.
   tags: Joi.array().items(Joi.objectId().meta({ ref: 'Tag' })),
   // Keywords are for SEO optimization and search engines.
-  keywords: Joi.array().items(Joi.string()),
-});
-
-const ArticleSchema = joiToMongoose(joiArticleSchema, {
-  usePushEach: true,
-});
-
-// TODO: replace with joi?
-ArticleSchema.pre('validate', function(next) {
-  if (this.type === 'video') {
-    ['video.platform', 'video.videoId'].forEach(path => {
-      if (!get(this, path)) {
-        next(new ValidationError(`Missing ${path} field for Video Article type`));
-      }
-    });
-    if (!VIDEO_PLATFORMS_LIST.includes(this.video.platform)) {
-      next(new ValidationError('video platform is not supported'));
-    }
-    if (!VIDEO_PLATFORMS[this.video.platform](this.video.videoId)) {
-      next(new ValidationError('bad videoId for selected platform'));
-    }
-  }
-  if (this.type === 'text' && this.video) {
-    next(new ValidationError('video must be absent if article type is text'));
-  }
-  next();
-});
-
-const IMAGES_SCHEMA = {
-  text: Joi.object({
-    page: Joi.image(),
-    horizontal: Joi.image(),
-    vertical: Joi.image(),
-  }),
-  video: Joi.object({
-    page: Joi.image(),
-    horizontal: Joi.image(),
-  }),
+  keywords: Joi.string(),
```

Comment:
why to refactor from array to a string?

Reply:
in metatag we end up with string, as well as text field in fibery

Reply:
still this is potentially a place lacking flexibility.
doesn't really matter, I am fine with any approach